File: backend/migrate.js
const { db } = require('./config/database');

function migrateOTPTable() {
  try {
    console.log('🔄 Starting OTP table migration...');
    
    // Check if is_verified column exists
    const tableInfo = db.prepare("PRAGMA table_info(otps)").all();
    const hasVerifiedColumn = tableInfo.some(col => col.name === 'is_verified');
    
    if (!hasVerifiedColumn) {
      console.log('➕ Adding is_verified column to otps table...');
      db.prepare('ALTER TABLE otps ADD COLUMN is_verified INTEGER DEFAULT 0').run();
      console.log('✅ Migration completed: is_verified column added successfully');
      
      // Show table structure after migration
      console.log('📋 Updated table structure:');
      const updatedTableInfo = db.prepare("PRAGMA table_info(otps)").all();
      updatedTableInfo.forEach(col => {
        console.log(`   - ${col.name}: ${col.type} ${col.dflt_value ? `(default: ${col.dflt_value})` : ''}`);
      });
    } else {
      console.log('✅ is_verified column already exists - no migration needed');
    }
    
    console.log('🎉 Migration process completed successfully!');
  } catch (error) {
    console.error('❌ Migration failed:', error.message);
    console.error('Full error:', error);
  }
}

// Run migration immediately
console.log('🚀 Lab Expert Database Migration');
console.log('================================');
migrateOTPTable();

module.exports = { migrateOTPTable };

File: py_backend/migrate.py
from sqlalchemy import text
from config.database import engine

with engine.connect() as conn:
    # Add online_status column to available_sensors table
    try:
        conn.execute(text("ALTER TABLE available_sensors ADD COLUMN online_status INTEGER DEFAULT 0"))
        conn.commit()
        print("Successfully added online_status column to available_sensors table")
    except Exception as e:
        if "duplicate column name" in str(e).lower() or "already exists" in str(e).lower():
            print("online_status column already exists, skipping...")
        else:
            print(f"Error adding online_status column: {e}")
            raise

print("Migration completed")